client/components/AllProducts.js

```diff
@@ -1,40 +1,48 @@
 import React, {Component} from 'react'
 import {connect} from 'react-redux'
 import {getAllProducts} from '../store/product'
-import {Link} from 'react-router-dom'
+import {NavLink} from 'react-router-dom'
 
 class AllProducts extends Component {
   componentDidMount() {
     this.props.getProducts()
   }
 
   render() {
-    // let products = this.props.products
-    // console.log(products)
+    const productList = this.props.products || []
+    console.log(this.props)
+
+    const productCard = productList.map(product => {
+      return (
+        <div className="col s12 m6 l3" key={product.id}>
+          <div className="card small">
+            <div className="card-image">
+              <img src={product.imageUrl} />
+            </div>
+            <div className="card-content">
+              <NavLink to={`/products/${product.id}`}>
```

Comment:
let's move the link to the product card level as opposed to only being on the product name